alpine-server/src/main/java/alpine/server/auth/OidcAuthenticationService.java

```diff
@@ -235,9 +235,10 @@ private OidcUser autoProvision(final AlpineQueryManager qm, final OidcProfile pr
         if (config.getPropertyAsBoolean(Config.AlpineKey.OIDC_TEAM_SYNCHRONIZATION)) {
             LOGGER.debug("Synchronizing teams for user " + user.getUsername());
             return qm.synchronizeTeamMembership(user, profile.getGroups());
+        } else {
+            // Only apply default teams during auto-provisioning, not on later updates:
+            return qm.addUserToTeams(user, config.getPropertyAsList(Config.AlpineKey.OIDC_TEAMS_DEFAULT));
```

Comment:
Doing it like this means that the default teams will be wiped upon next login, assuming team synchronization is enabled. So if the user logs out and in again, or simply switches to another browser / device, they will be back to no teams again.

There likely needs to be some additional logic in `synchronizeTeamMembership` that accounts for default teams.

Reply:
When team synchronization is enabled, the default teams are not taken into account at all.

The idea was that if you want team synchronization, you should manage your teams in your identity infrastructure. Only when you don't have team synchronization, you can use the default teams to give users initial/default authorizations.

Reply:
Ah, thanks for clarifying. I think I misunderstood what was being requested in #411 - I assumed it was about having a "catch-all" group for users who don't have any groups assigned yet in the IdP. Seems I missed the

> For organisations that are happy to manage users within an application

part of the original issue description. Makes sense to me then.